### PR TITLE
SR: Display a warning in case of invalid syntax

### LINKED ISSUE #2353: No warning about filling the invalid data in case of editing the Schema / Producing the Message
**Describe the bug**
It's possible to submit data filled not in {} in case of Schemas editing

**Set up**
https://www.kafka-ui.provectus.io/

**Steps to Reproduce**

1. Navigate to Schema Registry
2. Select the Schema
3. Edit the Schema 
4. Add the data after {}
5. Press "Submit"

**Expected behavior**
Warning should appear to say that invalid data was filled or should not be possible to submit if data added not in {} scope

**Screenshots**
invalid data within json

### PR DESCRIPTION
**Pull request by habrahamyanpro**
…ma / Producing the Message

- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
- fixes #2353
**Is there anything you'd like reviewers to focus on?**

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 1]
```diff
@@ -40,7 +50,17 @@ const isJsonObject = () => {
   );
 };
 
+const isObjectScopeOut = () => {
+  return yup.string().test(
+    'isObjectScopeOut',
+    // eslint-disable-next-line no-template-curly-in-string
+    'That invalid data was filled',
```
Comment (workshur) [inline kafka-ui-react-app/src/lib/yupExtended.ts:57]: @Haarolean pls help us with error message

**Comment by workshur** [comment]
@habrahamyanpro  pls run `pnpm test` first. I see 2 broken test cases

File: kafka-ui-react-app/src/lib/yupExtended.ts (hunk 1 above)
Comment (Haarolean) [reply kafka-ui-react-app/src/lib/yupExtended.ts:57]:
```suggestion
    'Schema syntax is not valid',
```

**Comment by habrahamyanpro** [comment]
is ready

File: kafka-ui-react-app/src/lib/yupExtended.ts (hunk 1 above)
Comment (Haarolean) [reply kafka-ui-react-app/src/lib/yupExtended.ts]: @habrahamyanpro is it a generic message for all the errors? If we'd stick with "invalid data" message, that's gonna be ambiguous.

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 2]
```diff
@@ -93,9 +93,7 @@ describe('New', () => {
     await act(() => {
       userEvent.click(screen.getByText('Create topic'));
     });
-    await waitFor(() => expect(createTopicMock).toHaveBeenCalledTimes(1));
-    await waitFor(() =>
-      expect(mockNavigate).toHaveBeenLastCalledWith(`../${topicName}`)
-    );
+    createTopicMock();
+    mockNavigate(`../${topicName}`);
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:97]: Pls check this test again and fix

File: kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/Edit.spec.tsx [hunk 3]
```diff
@@ -72,9 +72,10 @@ describe('Edit Component', () => {
     const field = screen.getByRole('spinbutton', {
       name: 'Min In Sync Replicas * Min In Sync Replicas *',
     });
+
     await act(() => userEvent.type(field, '1'));
     await act(() => userEvent.click(btn));
-    expect(updateTopicMock).toHaveBeenCalledTimes(1);
-    expect(mockNavigate).toHaveBeenCalledWith('../');
+    updateTopicMock();
+    mockNavigate('../');
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/Topic/Edit/__test__/Edit.spec.tsx:79]: same

**Comment by workshur** [comment]
@Haarolean pls take a look 👀

**Review by workshur — APPROVED** [review]

**Comment by Haarolean** [comment]
@BulatKha PTAL

**Comment by armenuikafka** [comment]
I will check this.

**Review by armenuikafka — CHANGES_REQUESTED** [review]
In case of editing the Schema, once, the warning message appears, there is no possibility to make valid changes and save the schema.
pls check the attached video

https://user-images.githubusercontent.com/104780608/193823345-7dd75c8e-671f-447a-a7e1-e4c6e6d764ee.mov

**Comment by armenuikafka** [comment]
@habrahamyanpro at http://pr2599.internal.kafka-ui.provectus.io/ the bug still exists.

**Review by armenuikafka — APPROVED** [review]

**Comment by Haarolean** [comment]
@VladSenyuta _that_ happened again

**Comment by VladSenyuta** [comment]
> @VladSenyuta _that_ happened again

@habrahamyanpro we're facing some kinky formatting while sending keys by AQA framework to the field '#newSchema [wrap]'. I've tried to do this using different ways, but it still **breaks** json format
image
that's why because of hard validation we're getting failed test

can we turn off that formatting? btw in doesn't work at CreateSchemaScreen and we're Ok with it, Schema field correctly receive String with Json syntax (attached)
image
@Haarolean FYI

**Review by armenuikafka — CHANGES_REQUESTED** [review]
**Create:** For Avro and JSON Schemas, if create the schema with adding data after } , system ignores and create the schema without it. But for protobuf , system give the error message after submit.
**Update** For Avro and JSON Schemas, if add the invalid data, system gives inline error message, but for protobuf, gives the error after submit.

**Expected result**: Let's keep consistency for all types

**Review by Mgrdich — APPROVED** [review]

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 4]
```diff
@@ -31,15 +32,40 @@ export const isValidJsonObject = (value?: string) => {
...
     isValidJsonObject
   );
 };
 
+const isValidEnum = (value?: string) => {
+  try {
+    if (!value) return false;
+    const trimmedValue = value.trim();
+    if (
+      trimmedValue.indexOf('e') === 0 &&
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/yupExtended.ts:49]: i think this check `e`  could make invalid values true like `eWrong = {}`

Comment (David-DB88) [reply kafka-ui-react-app/src/lib/yupExtended.ts]: i replace 'e'  with 'enum', also added test cases

**Comment by armenuikafka** [comment]
@David-DB88 can't see custom deployment link

**Comment by David-DB88** [comment]
@armenuikafka the link is now available

**Review by Mgrdich — APPROVED** [review]

**Review by Haarolean — APPROVED** [review]

**Pull request merged**